File: src/gatling/scala/org/openapitools/client/model/Pagination.scala

package org.openapitools.client.model


case class Pagination (
    /* Current index page returned */
    _currentPage: Option[Integer],
    /* Number of items per page returned */
    _size: Option[Integer],
    /* Total number of pages available */
    _totalPages: Option[Integer],
    /* Total number of elements (items) available */
    _totalElements: Option[Integer],
    _sort: Option[List[Sort]]
)
object Pagination {
    def toStringBody(var_currentPage: Object, var_size: Object, var_totalPages: Object, var_totalElements: Object, var_sort: Object) =
        s"""
        | {
        | "currentPage":$var_currentPage,"size":$var_size,"totalPages":$var_totalPages,"totalElements":$var_totalElements,"sort":$var_sort
        | }
        """.stripMargin
}

File: client/php-dt/src/App/DTO/Pagination.php
<?php
declare(strict_types=1);

namespace App\DTO;

use Articus\DataTransfer\Annotation as DTA;

class Pagination
{
    /**
     * Current index page returned
     * @DTA\Data(field="currentPage", nullable=true)
     * @DTA\Validator(name="Scalar", options={"type":"int"})
     */
    public ?int $current_page = null;

    /**
     * Number of items per page returned
     * @DTA\Data(field="size", nullable=true)
     * @DTA\Validator(name="Scalar", options={"type":"int"})
     */
    public ?int $size = null;

    /**
     * Total number of pages available
     * @DTA\Data(field="totalPages", nullable=true)
     * @DTA\Validator(name="Scalar", options={"type":"int"})
     */
    public ?int $total_pages = null;

    /**
     * Total number of elements (items) available
     * @DTA\Data(field="totalElements", nullable=true)
     * @DTA\Validator(name="Scalar", options={"type":"int"})
     */
    public ?int $total_elements = null;

    /**
     * @DTA\Data(field="sort", nullable=true)
     * @DTA\Strategy(name="Object", options={"type":\App\DTO\Collection1::class})
     * @DTA\Validator(name="TypeCompliant", options={"type":\App\DTO\Collection1::class})
     */
    public ?\App\DTO\Collection1 $sort = null;

}
